extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -425,6 +425,34 @@ export class ProjectsController {
 		await vscode.commands.executeCommand(constants.revealFileInOsCommand, vscode.Uri.file(project.projectFilePath));
 	}
 
+	/**
+	 * Opens the .sqlproj file for the given project. Upon update of file, prompts user to
+	 * reload their project.
+	 * @param context a treeItem in a project's hierarchy, to be used to obtain a Project
+	 */
+	public async editProjectFile(context: BaseProjectTreeItem): Promise<void> {
+		const project = this.getProjectFromContext(context);
+
+		try {
+			await vscode.commands.executeCommand(constants.vscodeOpenCommand, vscode.Uri.file(project.projectFilePath));
+			const projFileWatcher: vscode.FileSystemWatcher = vscode.workspace.createFileSystemWatcher(project.projectFilePath);
+
+			projFileWatcher.onDidChange(async (projectFileUri: vscode.Uri) => {
+				const result = await vscode.window.showInformationMessage(constants.refreshProject, constants.yesString, constants.noString);
+
+				if (result === constants.yesString) {
+					// Is there a better way to refresh a project with a changed .sqlproj file than close
+					//	and open project again?
+					this.projects = this.projects.filter((e) => { return e.projectFilePath !== projectFileUri.fsPath; });
```

Comment:
If you don't want to go the `readProjFile()` route, it's probably best to use `closeProject(context)` to avoid code duplication on this front.

Reply:
yeah close and reopen should be fine.

Reply:
I think my preference would be fore `readProjFile()` since closing and reopening would move the reloaded project to the bottom of the tree (filtering it out, then opening which does an append).

Reply:
oh we cant reopen and insert in the list in middle?

Reply:
Yup, I don't believe we can reopen and insert in the middle at the moment. My concerns with readProjFile() were:
- The other steps handled when opening a new project would be skipped: open reference projects, update for round tripping, read datasources.json. Not sure if those steps are needed after updating .sqlproj file?
- readProjFile() doesn't appear to be designed to run twice on a single project; would probably require some significant changes, as @Benjin mentioned

I'm thinking we could pursue one of these paths?
1. Updating readProjFile() to be able to run twice
2. Adding the ability to open new project at a specific location

Reply:
I don't think there will be "significant" changes to the `Project` class, just adding a `reset()` function that zeros out the various properties, and calling that at the beginning of `readProjFile()`.  I say the fact that it's not currently idempotent to read from a project file is an oversight on my part when I first wrote the function.